### ISSUE TITLE
tlayer crashes on native

### ISSUE THREAD

**Issue by benpicco** [des-testbed/RIOT-projects#43]
It doesn't matter which commands were run before, it crashes with and without init on

```
> send_udp
send_udp
cdab:0000:0000:0000:1236:ff00:00fe:0000
Start power: 0.000000
Used power: 0.000000
Start: 3278134751, End: 3278134752, Total: 1
Gleitkomma-Ausnahme (Speicherabzug geschrieben)
```

**Comment by LudwigKnuepfer** [comment]
@OlegHahm Didn't you fix this the day before yesterday? Which PR?
My guess would be https://github.com/RIOT-OS/RIOT/pull/256

**Comment by LudwigKnuepfer** [comment]
BTW, this really belongs in https://github.com/RIOT-OS/RIOT/issues/ .. reopen there if it's not already solved.

**Comment by OlegHahm** [comment]
Nope, it's just the transmission time calculation (division by zero).

This should fix it:

```
diff --git a/tlayer/tlayer_shell.c b/tlayer/tlayer_shell.c
index a0306a1..8486e36 100644
--- a/tlayer/tlayer_shell.c
+++ b/tlayer/tlayer_shell.c
@@ -358,7 +359,12 @@ void send_udp(char *str)
     printf("Used power: %f\n", ltc4150_get_total_Joule());
     printf("Start: %lu, End: %lu, Total: %lu\n", start.microseconds, end.microseconds, total.microseconds);
     secs = total.microseconds / 1000000;
-    printf("Time: %lu seconds, Bandwidth: %lu byte/second\n", secs, (count * 48) / secs);
+    if (!secs) {
+        puts("Transmission in no time!");
+    }
+    else {
+        printf("Time: %lu seconds, Bandwidth: %lu byte/second\n", secs, (count * 48) / secs);
+    }
     destiny_socket_close(sock);
 }

```

However, this project needs a few more changes to have it working on native. I'll make a PR later.

**Comment by LudwigKnuepfer** [comment]
Oh, ic, so it does belong here after all ;-)

**Comment by LudwigKnuepfer** [comment]
resolved by #46
